### PR TITLE
SONARIAC-666 S4830 Server certificates should be verified for curl/wget connections

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

**Comment by rudy-regazzoni-sonarsource** [comment]
Introducing method `withOptionalRepeating()` in `CommandDetector`.
This help to ignore multiple option until we find the proper one, `withOptional` was only permitting us to ignore only one element.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/UnsecureConnectionCheck.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import org.sonar.iac.docker.symbols.ArgumentResolution;
+import org.sonar.iac.docker.tree.api.RunInstruction;
+
+@Rule(key = "S4830")
+public class UnsecureConnectionCheck implements IacCheck {
+
+  private static final String MESSAGE = "Disabling TLS certificate verification is security-sensitive.";
+
+  private static final Set<String> SENSITIVE_CURL_OPTION = Set.of("-k", "--insecure", "--proxy-insecure", "--doh-insecure");
+  private static final CommandDetector SENSITIVE_CURL_COMMAND = CommandDetector.builder()
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/UnsecureConnectionCheck.java:39]:
```suggestion
  
  private static final CommandDetector SENSITIVE_CURL_COMMAND = CommandDetector.builder()
```

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/UnsecureConnectionCheck.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    .with("curl"::equals)
+    .withOptionalRepeating(s -> !SENSITIVE_CURL_OPTION.contains(s) && s.startsWith("-"))
+    .with(SENSITIVE_CURL_OPTION::contains)
+    .build();
+  private static final CommandDetector SENSITIVE_WGET_COMMAND = CommandDetector.builder()
```
Comment (nils-werner-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/UnsecureConnectionCheck.java:44]:
```suggestion
  
  private static final CommandDetector SENSITIVE_WGET_COMMAND = CommandDetector.builder()
```
For more readability?

**Review by nils-werner-sonarsource — APPROVED** [review]
> Introducing method withOptionalRepeating() in CommandDetector.

I'm currently, implementing something similar. It will not change the API. Maybe we can discuss which solution under the hood we prefer. However, this should not block the rule implementation. 

So far LGTM!

**Pull request merged**